tests/lib/state.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+SNAPD_STATE_PATH="$SPREAD_PATH"/snapd-state
+SNAPD_STATE_FILE="$SPREAD_PATH"/snapd-state.tar
+
+# shellcheck source=tests/lib/dirs.sh
+. "$TESTSLIB/dirs.sh"
+
+# shellcheck source=tests/lib/boot.sh
+. "$TESTSLIB/boot.sh"
+
+delete_snapd_state() {
+    if [[ "$SPREAD_SYSTEM" == ubuntu-core-16-* ]]; then
+        rm -rf "$SNAPD_STATE_PATH"
+    else
+        rm -f "$SNAPD_STATE_FILE"
+    fi
+}
+
+save_snapd_state() {
+    if [[ "$SPREAD_SYSTEM" == ubuntu-core-16-* ]]; then
+        boot_path="$(get_boot_path)"
+        test -n "$boot_path" || return 1
+
+        mkdir -p "$SNAPD_STATE_PATH" "$SNAPD_STATE_PATH"/system-units
+
+        # Copy the state preserving the timestamps
+        cp -a /var/lib/snapd "$SNAPD_STATE_PATH"/snapd-lib
+        cp -rf "$boot_path" "$SNAPD_STATE_PATH"/boot
+        cp -f /etc/systemd/system/snap-*core*.mount "$SNAPD_STATE_PATH"/system-units
+    else
+        escaped_snap_mount_dir="$1"
```

Comment:
Could we computed that ourselves or alternatively require that it is always passed. A function that _sometimes_ requires an argument is more error prone (especially in shell)

Reply:
done